### DISCUSSION TITLE
Effect of on:click handlers on Client side Navigation

### DISCUSSION THREAD

**Discussion by EmilienKopp** [sveltejs/kit#12702] [Q&A] [answered]
Just to make sure I got everything right before submitting a PR to the documentation, but am I correct in my understanding that any `<a href=... >` tag with an on:click handler will cause Sveltekit to ignore client side navigation altogether?

I had one `on:click|stopPropagation` on one of my links that caused me to pull my hair for a few days trying to understand why my whole tree of load functions would rerun though I had no invalidating behavior in any of them. Removing it did seem to solve the issue, but I feel like this would deserve a small warning in the docs.

**Comment by brunnerh** [comment, marked as answer]
> but am I correct in my understanding that any `<a href=... >` tag with an on:click handler will cause Sveltekit to ignore client side navigation altogether?

That is not the case, but if you add `stopPropagation` that will of course mess with any mechanism that relies on the event bubbling up the tree. Client-side navigation appears to be one such mechanism.

**Reply by EmilienKopp** [reply]
Thanks for the humbling clarification! It just makes so much sense and I feel stupid for not realizing, and now I realize that indeed onclick handlers alone never appeared to be causing such behavior.
